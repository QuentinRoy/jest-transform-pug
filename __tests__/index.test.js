test("Simple case", () => {
  const template = require("./templates/template-simple.pug");
  expect(template()).toMatchSnapshot();
});

test("With arguments", () => {
  const template = require("./templates/template-with-args.pug");
  expect(
    template({ url: "http://example.com", title: "Test Title" }),
  ).toMatchSnapshot();
  expect(
    template({ url: "http://something-else.com", title: "Other title" }),
  ).toMatchSnapshot();
});

test("With include", () => {
  const template = require("./templates/template-with-include.pug");
  expect(template()).toMatchSnapshot();
});

test("With extends", () => {
  const template = require("./templates/template-with-extends.pug");
  expect(template()).toMatchSnapshot();
});
