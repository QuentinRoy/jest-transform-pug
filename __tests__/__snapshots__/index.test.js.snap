// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple case 1`] = `"<!DOCTYPE html><html><head><title>My Page</title></head><body><div id="my-div"><p>Hello, World!</p></div></body></html>"`;

exports[`With arguments 1`] = `"<!DOCTYPE html><html><head><title>Test Title</title></head><body><a href="http://example.com">A link</a></body></html>"`;

exports[`With extends 1`] = `"<!DOCTYPE html><html><head><title>Test</title><script src="/jquery.js"></script><script src="/pets.js"></script></head><body><h1>Title</h1><ul><li>cat</li><li>dog</li></ul><div id="footer"><p>some footer content</p></div></body></html>"`;

exports[`With include 1`] = `"<!DOCTYPE html><html><head><title>My Site</title><script src="/javascripts/jquery.js"></script><script src="/javascripts/app.js"></script></head><body><h1>Test</h1><p>Welcome to my super lame test.</p></body></html>"`;
